### PR TITLE
Rpr 48

### PR DESCRIPTION
**Pull request by linorabolini**
Hello !

Initial commit for the Zeus Core module.
Added tests and added plugin support.
Migrated some of the functions I found in the googleTag init function to plugins.

One big thing missing is the ad refresh logic, but thats bigger and will tackle it in RPR-52.

I have been trying to decouple the logic from the global scope but many functions across the code access different variables stored in window.wpAd so It may take a bit longer than expected to decouple all and modularize :(

I didn't add flow to these modules because to me it is not clear yet that this is going to be the final interface. I have been checking the wp-ad code and im brainstorming how to integrate it. I'll be doing some testing on this area to evaluate the best approach! Ideas are welcomed !

### CODE REVIEW

File: src/module/adnode/index.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+function render(node) {
+    if (!node.slot) {
+        const msg = `Couldn't render the closed spot: ${node.id}`
+        return core.warn(msg)
+    }
+    wpAd.exeProgrammticProcess([this])
+}
+
+// TODO: move this to the core
+function buildGPT(node, googletag) {
```
[the remark this answers is not in the document]
Comment (linorabolini) [reply src/module/adnode/index.js]: Hey Aram ! 

Will we want to support other types of ad servers apart from google tag? The main part of the zeus flow is built on top of google tag events and async queue, but if that's true I can just build an interface that the other ad server types will need to follow so we can plug in the one we want :)

[the remark this answers is not in the document]
Comment (linorabolini) [reply src/module/adnode/index.js]: oh and yes, I renamed these functions into 2 single functions:

createSlot and destroySlot, so its simple and clear and we do not have it spread in the old steps: buildGPT(), getGPTSlot() and GPTSlot().

[the remark this answers is not in the document]
Comment (eric-washpost) [reply src/module/adnode/index.js]: There is a feature that we can load ad from the Ad CMS for direct sold campaign or in-house ads. Also we may pull ad from other ad servers on the future, but we are focusing on the google tag for now.

File: src/module/adnode/index.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    return googletag
+        .pubads()
+        .getSlots()
+        .find(slot => slot.getSlotElementId() === node.id)
+}
+
+// TODO: move this to the core
+function getOrCreateGTPSlot(node, googletag) {
```
[the remark this answers is not in the document]
Comment (linorabolini) [reply src/module/adnode/index.js]: just as a note, when moving this to the core renderer logic in #RPR-45, I simplified all GPT functions into just 2. 
createSlot and destroySlot :)

**Pull request merged**